src/Callout.jsx

```diff
@@ -0,0 +1,262 @@
+var React = require('react');
```

Comment:
My challenge with having the callout contain the text area is we are baking in such a deep dependency on the text area that it could never be used on much else - like the editor for instance.

I wonder if perhaps we could make it more modular like:
```javascript
<TextField 

   // This same property could be used for editors later
   autoComplete={true} 

    // Optional callback for transforming words like '@sky' into '[user:123] '
   autoCompleteTransform={ function(text) { return text; } }
   
   // This is the same as suggestions you used before with raw data
   autoCompleteData={ suggestions }
   
   // As an alternate to suggestions, this callback could be used for ajax server requests
   autoCompleteGetData={function(doneCallback) { doneCallback(suggestions) } }
   
   // If the trigger is null, then it should start autocomplete after two chars are entered
   // It can also have multiple triggers for different kinds of data like adding colors with #,
   // and user callouts with '@'
   autocompleteTrigger={["@"]}

   // Delimiter will give us the ability to make it ';' or ',' for things like email input but defaults to space/words
   autocompleteDelimiter={[" "]}
/>
```

If we do it like that then the above could later be added to <Editor autoComplete={true} autoCompleteData={suggestions} /> and so on.